src/components/CustomerQuota/CheckoutSuccess/CheckoutSuccessCard.tsx

```diff
@@ -40,22 +48,109 @@ const UsageQuotaTitle: FunctionComponent<{
   </>
 );
 
+export interface TransactionsByTimeMap {
+  [transactionTimeInSeconds: string]: {
+    transactionTime: Date;
+    transactions: Transaction[];
+    order: number;
+  };
+}
+
+/**
+ * Given past transactions, group them by timestamp
+ * If less than 1 second apart, count as same group.
+ *
+ * @param sortedTransactions Past transaction results sorted by transaction time in desc order
+ * @param allProducts Policies
+ */
+export const groupTransactionsByTime = (
+  sortedTransactions: PastTransactionsResult["pastTransactions"] | null,
+  allProducts: CampaignPolicy[]
+): TransactionsByTimeMap => {
+  const transactionsByTimeMap: {
+    [transactionTimeInSeconds: string]: {
+      transactionTime: Date;
+      transactions: Transaction[];
+      order: number;
+    };
+  } = {};
+  sortedTransactions?.forEach(item => {
+    const policy = allProducts?.find(
+      policy => policy.category === item.category
+    );
+    const categoryName = policy?.name ?? item.category;
+    const transactionTimeInSeconds = String(
+      Math.floor(item.transactionTime.getTime() / 1000)
+    );
+
+    if (!(transactionTimeInSeconds in transactionsByTimeMap)) {
+      transactionsByTimeMap[transactionTimeInSeconds] = {
+        transactionTime: item.transactionTime,
+        transactions: [],
+        order: -transactionTimeInSeconds
```

Comment:
Nice use of the negation to reverse the sort. Does this imply the policy `order` should also allow negative numbers?

Reply:
Hmm I think it is a separate consideration. Since that is manually set, not sure when we might need negative numbers~

Reply:
Okay got it, unrelated to the policy `order` then!